### PR TITLE
[feature] apis of edge application

### PR DESCRIPTION
**Pull request by vincentgoat**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Implementation of feature at #3582
This pr includes
1. edge application APIs(apis/client)
2. update generate-crds.sh and update-codegen.sh

/cc @fisherxu @Congrool @lonelyCZ

### CODE REVIEW

**Comment by zhu733756** [comment]
Good job!

File: pkg/apis/grouping/v1alpha1/register.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package v1alpha1 contains API Schema definitions for the grouping v1alpha1 API group
+// +k8s:openapi-gen=true
+// +k8s:deepcopy-gen=package,register
+// +k8s:conversion-gen=kubeedge/cloud/pkg/apis/grouping
```
[the remark this answers is not in the document]
Comment (Congrool) [reply pkg/apis/grouping/v1alpha1/register.go:20]: It seems that it should be `kubeedge/pkg/apis/grouping`.

By the way, we seem not to generate conversion and defaulter code for this type, and others have been provided at `doc.go`. So, how about removing these tags from register.go?

[the remark this answers is not in the document]
Comment (vincentgoat) [reply pkg/apis/grouping/v1alpha1/register.go:20]: thanks, I will update it

**Comment by vincentgoat** [comment]
close this pr instead by #3719

**Pull request closed without merging**